flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -56,6 +56,23 @@
 
 #define DEBUG_TYPE "flang-lower-expr"
 
+static mlir::Type genRealType(mlir::MLIRContext *context, int width) {
```

Comment:
I don't think you have to re-derive these from bitwidths. There are functions in ConvertType.cpp that do all the necessary lifting based on type category and kind values.